.github/workflows/pytest.yaml

```diff
@@ -6,22 +6,17 @@ jobs:
   pytest:
     name: Run tests
     runs-on: ubuntu-latest
+    container:
+      image: python:3.12-bullseye
 
     steps:
       - name: Checkout repo
         uses: actions/checkout@v4
 
-      - name: Setup Python
-        uses: actions/setup-python@v5
-        with:
-          python-version: '3.12'
-          cache: 'pip'
-
       - name: Install system dependencies
         run: |
-          # dont run update, it is slow 
-          # sudo apt-get update
-          sudo apt-get install -y --no-install-recommends \
+          apt-get update && \
```

Comment:
Has this stopped working without running the update? If not, can we keep it disabled? There is nothing essential in the updates to run `pytest`, so it needlessly increases the action running time.

Reply:
Yes, this has stopped working. The Python docker images don't have the package metadata cached because it's usually just wasted disk space so if you do need to install something inside that docker container you will need to run apt update first. 

Theoretically you could decide to run pytest on ubuntu-latest like before (which would fix that issue), but then you would test on a different OS than you do the actual compiling / bundling on, not sure if that's a good idea.

Reply:
I understand the rationale behind the change, but slowing down the test suite (which runs quite frequently) feels unreasonable when the actual change you want to achieve is relevant for the artifact build. 

Considering we're building also macOS and Windows versions, which dont run the testsuite at all and we still trust the build will be okay, then running *Python* tests on different version than where the build happens should be okay as well.

Reply:
Okay, then I'll revert that step and have the tests run on a newer version.

Reply:
I have reverted the changes to pytest.yaml which should fix both of your concerns, now it's just the release.yml that has changed.